analyze-scope.js

```diff
@@ -286,6 +300,41 @@ class Referencer extends OriginalReferencer {
         this.typeMode = upperTypeMode;
     }
 
+    /**
+     * Visit new expression.
+     * @param {NewExpression} node The NewExpression node to visit.
+     * @returns {void}
+     */
+    NewExpression(node) {
+        this.visitTypeParameters(node);
+        this.visit(node.callee);
+        if (node.arguments) {
+            node.arguments.forEach(this.visit, this);
+        }
+    }
+
+    /**
+     * Override.
+     * Visit call expression.
+     * @param {CallExpression} node The CallExpression node to visit.
+     * @returns {void}
+     */
+    CallExpression(node) {
+        this.visitTypeParameters(node);
+
+        // Check this is direct call to eval
+        if (!this.scopeManager.__ignoreEval() && node.callee.type === "Identifier" && node.callee.name === "eval") {
```

Comment:
We can remove this block because `ignoreEval` is always true ([analyze-scope.js#L599](https://github.com/eslint/typescript-eslint-parser/blob/491ebcaff9815071d4194584e3d94d0edf601e3b/analyze-scope.js#L599)) :smiley:

Reply:
awsome